### PR TITLE
Update to latest WebGPU WebIDL

### PR DESCRIPTION
**Pull request by grovesNL**
Update to latest WebGPU WebIDL as of November 7.

Like in #2267, there are some temporary changes from the upstream WebIDL for Gecko:
- Define GPUCode as typedef (USVString or Uint32Array) GPUCode; to allow both SPIR-V and WGSL for now
- Continue to use the old signature for set_index_buffer because that's what Gecko is currently using

I have the latest rustfmt as far as I can tell, but regenerating the bindings seems to shift things around by adding extra spaces in attributes. Is there a specific version of rustfmt I should be using? I currently have a formatting commit but I'll remove it if I can fix this.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks! According to CI though it looks like your local rustfmt may be doing something differently? Maybe a global config file? We should just be using the default rustfmt settings for the stable channel to format everything

File: crates/web-sys/Cargo.toml [hunk 1]
```diff
@@ -632,7 +634,7 @@ IirFilterOptions = []
 ImageBitmap = []
 ImageBitmapFormat = []
 ImageBitmapRenderingContext = []
-ImageCapture = ["EventTarget"]
+ImageCapture = []
```
Comment (grovesNL) [inline crates/web-sys/Cargo.toml:637]: (this wasn't related to the WebGPU changes, but it was removed when I regenerated the bindings)

**Comment by grovesNL** [comment]
Thanks for taking a look! Yeah I noticed the CI check failing and looked into this a bit more. I freshly cloned the repository and the formatting issue disappeared, so it turns out it wasn't related to rustfmt at all.

Instead it seems to be caused by the output of proc-macro2 when it's formatting the attribute initially. I had an older version in my Cargo.lock, so I updated proc-macro2 (`cargo update -p proc-macro2`) and the formatting is alright again. It was probably a small change in the proc-macro2 output (maybe alexcrichton/proc-macro2/pull/246).

I think everything should be alright now.

**Comment by alexcrichton** [comment]
Ah right yes forgot about that!

**Pull request merged**